pkg/api/v1/handlers/rpc.go

```diff
@@ -0,0 +1,162 @@
+// Package handlers provides HTTP request handling
+package handlers
+
+import (
+	"encoding/json"
+
+	"github.com/celestiaorg/talis/internal/db/models"
+	"github.com/celestiaorg/talis/internal/services"
+
+	fiber "github.com/gofiber/fiber/v2"
+)
+
+// RPCRequest defines the structure for RPC-style API requests
+type RPCRequest struct {
+	// Method is the operation to perform (e.g., "project.create", "task.list")
+	Method string `json:"method"`
+
+	// Params contains the operation parameters
+	Params interface{} `json:"params"`
+
+	// ID is an optional request identifier that will be echoed back in the response
+	ID string `json:"id,omitempty"`
+}
+
+// RPCResponse defines the structure for RPC-style API responses
+type RPCResponse struct {
+	// Data contains the operation result
+	Data interface{} `json:"data,omitempty"`
+
+	// Error contains error information if the operation failed
+	Error *RPCError `json:"error,omitempty"`
+
+	// ID echoes back the request ID if provided
+	ID string `json:"id,omitempty"`
+
+	// Success indicates if the operation was successful
+	Success bool `json:"success"`
+}
+
+// RPCError defines the structure for RPC errors
+type RPCError struct {
+	// Code is a numeric error code
+	Code int `json:"code"`
+
+	// Message is a human-readable error message
+	Message string `json:"message"`
+
+	// Data contains additional error details (optional)
+	Data interface{} `json:"data,omitempty"`
+}
+
+// RPCHandler handles RPC-style API requests for projects and tasks
+type RPCHandler struct {
+	projectService *services.ProjectService
+	taskService    *services.TaskService
+}
+
+// RPCServices contains the services needed by the RPC handler
+type RPCServices struct {
+	ProjectService *services.ProjectService
+	TaskService    *services.TaskService
+}
```

Comment:
copy paste error? These are the same

Reply:
We might consolidate them into one struct eventually. I used this way in order not to have a long list of func args.